### PR TITLE
Add globus_token to get_assaytype_data

### PR DESCRIPTION
**Pull request by gesinaphillips**
- Add globus_token to `validation_utils.get_assaytype_data`
- Update tests
- Switch from concatenating URLs to urllib methods

### CODE REVIEW

File: src/ingest_validation_tools/upload.py [hunk 1]
```diff
@@ -371,7 +378,8 @@ def _constraint_checks(self, schema: SchemaVersion):
             "Authorization": f"Bearer {self.globus_token}",
             "Content-Type": "application/json",
         }
-        url = f"{self.app_context['constraints_url']}?match=True&order={CONSTRAINTS_CHECK_METHOD}"
+        params = urlencode({"match": True, "order": CONSTRAINTS_CHECK_METHOD})
+        url = urljoin(self.app_context["constraints_url"], f"?{params}")
         response = requests.post(url, headers=headers, data=data)
```
Comment (jpuerto-psc) [inline src/ingest_validation_tools/upload.py:383]: You should be able to use the `params` parameter in the post call here:

https://requests.readthedocs.io/en/latest/user/quickstart/#passing-parameters-in-urls

So:
```
params = {"match": True, "order" CONSTRAINTS_CHECK_METHOD}
requests.post(self.app_context['constraints_url'], headers=headers, data=data, params=params)
```

File: src/ingest_validation_tools/upload.py [hunk 2]
```diff
@@ -627,6 +632,7 @@ def _check_url(
                     *get_entity_type_vals(response.json()),
                 )
         elif field in ["orcid_id", "orcid"]:
+            url = urljoin(constrained_fields[field], f"?q=orcid:{value}")
             headers = {"Accept": "application/json"}
             response = requests.get(url, headers=headers)
```
Comment (jpuerto-psc) [inline src/ingest_validation_tools/upload.py:637]: Same suggestion as above

File: src/ingest_validation_tools/upload.py (hunk 1 above)
Comment (gesinaphillips) [reply src/ingest_validation_tools/upload.py:383]: That's so much better!

File: src/ingest_validation_tools/upload.py (hunk 2 above)
Comment (gesinaphillips) [reply src/ingest_validation_tools/upload.py:637]: The params for the ORCID API are so hateful, that {"q": "orcid:{value}"} pattern threw me until literally right this second. Thank you for helping me give this a second look :)

**Review by jpuerto-psc — APPROVED** [review]

**Pull request merged**